src/common/history.c

```diff
@@ -1095,45 +1112,35 @@ void dt_history_compress_on_image(int32_t imgid)
     sqlite3_step(stmt);
     sqlite3_finalize(stmt);
   }
+  dt_pthread_mutex_unlock(&(darktable.db_image[imgid & (DT_IMAGE_DBLOCKS-1)]));
 }
 
-#define no_forced_reordering FALSE
-#define give_reorder_information FALSE
 static void _history_reorder(int32_t imgid)
 {
   int32_t dummy = 0x7fffffff;
   sqlite3_stmt *stmt;
-  if(give_reorder_information) fprintf(stderr,"\n\n_history reorder for image: %i",imgid);
 
-  if(no_forced_reordering)
-  {
-    if (give_reorder_information) fprintf(stderr,", no action\n");
-  }
-  else
-  {
-    if (give_reorder_information) fprintf(stderr,", reorder\n");
-    // make sure running jobs can't interfere here as the followiing code uses a fixed dummy id
-    // and also intends to have a "properly" orderered database
-    dt_pthread_mutex_lock(&darktable.db_insert);
+  // make sure running jobs can't interfere here as the followiing code uses a fixed dummy id
+  // and also intends to have a "properly" orderered database
+  dt_pthread_mutex_lock(&(darktable.db_image[imgid & (DT_IMAGE_DBLOCKS-1)]));
+  dt_pthread_mutex_lock(&(darktable.db_image[dummy & (DT_IMAGE_DBLOCKS-1)]));
```

Comment:
same ascending ordering must be applied here too

Reply:
or maybe not actually. since dummy & (DT_IMAGE_DBLOCKS-1) is a constant, it is already ordered lock. OK then.